src/models/event-timeline.js

```diff
@@ -339,6 +339,27 @@ EventTimeline.prototype.addEvent = function(event, atStart) {
     }
 };
 
+
+/**
+ * Replaces an event in the timeline, if it exists.
+ *
+ * @param {string} eventId  the id of the existing event
+ * @param {boolean} newEvent  the new event
+ * @return {MatrixEvent?} the old event, if found and replaced.
+ */
+EventTimeline.prototype.tryReplaceEvent = function(eventId, newEvent) {
+    const eventIndex = this._events.findIndex(event => event.getOriginalId() === eventId);
+    if (eventIndex === -1) {
+        return;
+    }
+    const oldEvent = this._events[eventIndex];
+    this._events[eventIndex] = newEvent;
+    // copy over metadata, as we don't hold state for random locations in the timeline, only for the end or start
+    // and assuming for now that replacing an event doesn't change the sender that should be shown.
+    newEvent.sender = oldEvent.sender;
```

Comment:
Will this work move into an `Event.replace(...)` method, since I believe you're planning to change to be more like redaction?

Reply:
Yes, I already have that almost working, apart from react not updating the tile for some reason. I'd be tempted to still merge this as it is, as it works and I'd prefer as much testing time on this as we can get.

Reply:
Right, seems okay to merge as is. I just wanted to make sure I understand the plan. 😄